src/eterna/ui/AnnotationLayersPanel.ts

```diff
@@ -522,36 +628,53 @@ export default class AnnotationLayersPanel extends ContainerObject {
             annotation.isSelected.connect((isSelected) => {
                 if (isSelected && annotation.type !== AnnotationItemType.CATEGORY) {
                     this._selectedPath = annotation.indexPath;
+                    this._selectedItem = annotation;
                     this._selectedDisplay = annotation.display;
 
-                    // Show delete button
-                    this.updateUpperToolbar(true);
+                    // Show delete + edit button
+                    const withEdit = Eterna.playerID === annotation.playerID
+                        && (
+                            annotation.category !== AnnotationCategory.PUZZLE
+                            || this._puzzleAnnotationsEditable
+                        )
+                        && annotation.category !== AnnotationCategory.STRUCTURE
+                        && annotation.type === AnnotationItemType.ANNOTATION;
+                    const withDelete = Eterna.playerID === annotation.playerID
+                    && (
+                        annotation.category !== AnnotationCategory.PUZZLE
+                        || this._puzzleAnnotationsEditable
+                    )
+                    && annotation.category !== AnnotationCategory.STRUCTURE;
+                    this.updateUpperToolbar(withDelete, withEdit);
```

Comment:
I'm not sure why you have to look at the user ID at all. Structure labels can never be edited, puzzle labels can be edited in puzzlemaker, and ad-hoc annotations can be added at any time

Reply:
Thanks for the heads up! I had initially included it there to foreshadow collaborative puzzle solving/making. We don’t have collaborative sessions in Eterna (maker or solver) right now so explicitly checking that a user created an annotation isn’t necessary. Will switch it up to a simpler condition